### PR TITLE
newdelet icon

### PR DESCRIPTION
**Pull request by Mitrovic-Milan**

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
 import styled from 'styled-components';
+import {Icon} from '@iconify/react';
 
-export const Button = ({textValue}) => {
-  return <StyledButton type="submit">{textValue}</StyledButton>;
+export const Button = ({textValue, onClick, dataid}) => {
+  return (
+    <StyledButton type="submit" onClick={onClick} data-id={dataid}>
+      {textValue === 'Delete' ? '' : textValue}
+      {textValue === 'Delete' ? <Icon icon="ant-design:delete-outlined" style={{fontsize: '28px'}} /> : ''}
```
Comment (stefankorder) [inline src/components/Button.js]: You do not need two different tenery's for this case. You can solve this with one single tenery.

File: src/components/Form.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-    setInput(e.target.value);
+    newInput(e.target.value);
   };
 
   const onFormSubmit = e => {
     e.preventDefault();
-    setTodos([...todos, {id: nanoid(), title: input, completed: false}]);
-    setInput('');
+    newTodo([...todos, {id: nanoid(), title: input, completed: false}]);
+    newInput('');
```
Comment (stefankorder) [inline src/components/Form.js]: The whole logic should be at the level your state is managed. Keep your components stupid as possible.

File: src/components/Form.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
 import {nanoid} from 'nanoid';
 import {Button} from './Button';
 
-const Form = ({input, setInput, todos, setTodos}) => {
+const Form = ({input, newInput, todos, newTodo}) => {
   const onInputChange = e => {
-    setInput(e.target.value);
+    newInput(e.target.value);
```
Comment (stefankorder) [inline src/components/Form.js]: The whole logic should be at the level your state is managed. Keep your components stupid as possible.

File: src/components/TodosList.js [hunk 4]
```diff
@@ -1,22 +1,44 @@
 import styled from 'styled-components';
+import {Button} from './Button';
+
+const TodosList = ({todos, deleteTodo}) => {
+  function onHandleClick(e) {
+    const todoid = e.target.dataset.id;
+    deleteTodo(todoid);
```
Comment (stefankorder) [inline src/components/TodosList.js]: The whole logic should be at the level your state is managed. Keep your components stupid as possible.

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work! Please take a look at my comments. If you need any help with the implementation please contact me.

File: src/util/LocalStorage.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const loadFromLocalStorage = key => {
+  const value = localStorage.getItem(key);
+  return JSON.parse(value);
```
Comment (actyralf) [inline src/util/LocalStorage.js]: `JSON.parse()` can fail and throw an error when called with invalid JSON data. You need to handle this situation with a `try/catch` block.

**Review by actyralf — CHANGES_REQUESTED** [review]

**Pull request merged**